### PR TITLE
clsuter: let fewer replicas region has high priority to scheduler

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?

issue: #3699

### What is changed and how it works?
check suspect and check waiting regions should be sorted by miss peer count 
### Check List

Tests

- Unit test

Code changes

Side effects

Related changes

- Need to cherry-pick to the release branch

### Release note

```release-note
let fewer replicas region has a high priority to the scheduler
```

### CODE REVIEW

**Review by nolouch — COMMENTED** [review]
Is it can pick to release 4.0?

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -454,8 +455,21 @@ func (c *RaftCluster) AddSuspectRegions(regionIDs ...uint64) {
 // GetSuspectRegions gets all suspect regions.
 func (c *RaftCluster) GetSuspectRegions() []uint64 {
 	c.RLock()
-	defer c.RUnlock()
-	return c.suspectRegions.GetAllID()
+	regions := c.suspectRegions.GetAllID()
+	c.RUnlock()
+	sort.Slice(regions, func(i, j int) bool {
```
Comment (nolouch) [inline server/cluster/cluster.go:460]: If the store is down, there may many regions.

**Comment by disksing** [comment]
When we put a region to the wait list, we can put some extra data or tag to explain the reason that a region needs to fix. In this way, we don't have to check again.

**Comment by bufferflies** [comment]
> When we put a region to the wait list, we can put some extra data or tag to explain the reason that a region needs to fix. In this way, we don't have to check again.

the pr will be close,release 4.0 do not have waiting queue .so I prepare devide the pr into two pr .  it is a good idea for check miss peer again.

**Pull request closed without merging**